azurelinuxagent/common/osutil/redhat.py

```diff
@@ -140,17 +139,83 @@ def get_nm_controlled(self):
 
         return True
 
-    def publish_hostname(self, hostname):
+    def get_nic_operational_and_general_states(self, ifname):
+        """
+        Checks the contents of /sys/class/net/{ifname}/operstate and the results of 'nmcli -g general.state device show {ifname}' to determine the state of the provided interface.
+        Raises an exception if the network interface state cannot be determined.
+        """
+        filepath = "/sys/class/net/{0}/operstate".format(ifname)
+        nic_general_state_cmd = ['nmcli', '-g', 'general.state', 'device', 'show', ifname]
```

Comment:
are these expected to be there and work in all versions?

Reply:
Yes, nmcli is the NetworkManager command line management tool for fedora based systems: https://fedoraproject.org/wiki/Tools/NetworkManager/CLI

If for some unexpected reason this command returns a non zero exit code, a CommandError will be raised and we'll raise it to the MonitorHostNameChanges operation which will report it to telemetry.